crates_io_cdn_logs/src/cloudfront.rs

```diff
@@ -0,0 +1,217 @@
+use crate::paths::parse_path;
+use crate::DownloadsMap;
+use chrono::NaiveDate;
+use std::borrow::Cow;
+use tokio::io::{AsyncBufRead, AsyncBufReadExt};
+use tracing::{instrument, warn};
+
+const HEADER_PREFIX: char = '#';
+const HEADER_VERSION: &str = "#Version:";
+const HEADER_FIELDS: &str = "#Fields:";
+
+const FIELD_DATE: &str = "date";
+const FIELD_METHOD: &str = "cs-method";
+const FIELD_PATH: &str = "cs-uri-stem";
+const FIELD_STATUS: &str = "sc-status";
+
+#[instrument(level = "debug", skip(reader))]
+pub async fn count_downloads(reader: impl AsyncBufRead + Unpin) -> anyhow::Result<DownloadsMap> {
+    let mut num_fields = 0;
+    let mut date_index = None;
+    let mut method_index = None;
+    let mut path_index = None;
+    let mut status_index = None;
+
+    let mut downloads = DownloadsMap::new();
+
+    let mut lines = reader.lines();
+    while let Some(line) = lines.next_line().await? {
+        if let Some(version) = line.strip_prefix(HEADER_VERSION) {
+            let version = version.trim();
+            if version != "1.0" {
+                anyhow::bail!("Unsupported version: {}", version);
+            }
+            continue;
+        }
+
+        if let Some(fields_str) = line.strip_prefix(HEADER_FIELDS) {
+            let fields = fields_str.trim().split(' ').collect::<Vec<_>>();
```

Comment:
It scares me a tiny bit that neither the Amazon documentation nor the (27 year old draft!) RFC it refers actually specifies the separator for field names, but obviously we know what it seems to be in practice. :grimacing:

Reply:
"whitespace" and yet it uses spaces in the header and tabs in the payload... 🙄